### DISCUSSION TITLE
Secondary windows for custom widgets

### DISCUSSION THREAD

**Discussion by KR155E** [eclipse-theia/theia#13570] [Q&A] [answered]
I have tried adding secondary window support to my custom widgets. It was pretty straight forward to get the base funtionality working by implementing ExtractableWidget. However, all my custom styles are missing in the secondary window and I am having trouble finding out what needs to be done to get those loaded. Do my CSS files have to be registered with some service or similar?

**Comment by msujew** [comment]
@tsmaeder If I remember correctly, you have been struggling with this as well. Can you recommend something?

**Comment by marcdumais-work** [comment]
Seems related but may not reflect current state of things: https://github.com/eclipse-theia/theia/issues/11643#issuecomment-1248157956

**Comment by tsmaeder** [comment, marked as answer]
If you add the following to your theia extensions, it will include the CSS reachable from the mentioned module in the secondary window:
```
"theiaExtensions": [
    {
      "secondaryWindow": "lib/browser/terminal-frontend-module",
      ...
```
From there, it's a question of your widget being able to live in one window, but having it's html in another: There are a couple of caveates, for example, you cannot use the global `window` object, since that is the original window, or the HTML Element classes are per Windows, so "instanceof" with the class name will not work for html elements in the secondary window. With a bit of luck, though, your widget will "just work".

**Reply by KR155E** [reply]
That did the trick! Thank you!

**Reply by tsmaeder** [reply]
Btw: if you're looking to go to EclipseCon FOR FREE: experience reports about using Theia would be a good topic for [EclipseCon](https://www.ocxconf.org/event/778b82cc-6834-48a4-a58e-f883c5a7b8c9/websitePage:77320887-9b93-4211-8865-637a602eb991)
